First, let's import the needed libraries: Pandas

In [25]:
import pandas as pd 

Load all the csv files and merge them into one dataframe

In [18]:
phy1 = pd.read_csv("/Users/Tatum/Downloads/Physical dataset/phy_att_1.csv", encoding="utf-16", sep="\t", engine="python")
phy2 = pd.read_csv("/Users/Tatum/Downloads/Physical dataset/phy_att_2.csv", encoding="utf-16", sep="\t", engine="python")
phy3 = pd.read_csv("/Users/Tatum/Downloads/Physical dataset/phy_att_3.csv", encoding="utf-16", sep="\t", engine="python")
phy4 = pd.read_csv("/Users/Tatum/Downloads/Physical dataset/phy_att_4.csv", encoding="utf-8-sig")
df = pd.concat([phy1, phy2, phy3, phy4], ignore_index=True)
df.head()

,Time,Tank_1,Tank_2,Tank_3,Tank_4,Tank_5,Tank_6,Tank_7,Tank_8,Pump_1,...,Valv_16,Valv_17,Valv_18,Valv_19,Valv_20,Valv_21,Valv_22,Label_n,Label,Lable_n
0,09/04/2021 18:23:28,0,0,0,0,0,0,0,0,False,...,False,False,False,False,False,False,False,0.0,normal,NaN
1,09/04/2021 18:23:29,0,0,0,0,0,0,0,0,False,...,False,False,False,False,False,False,False,0.0,normal,NaN
2,09/04/2021 18:23:30,0,0,0,0,0,0,0,0,False,...,False,False,False,False,False,False,False,0.0,normal,NaN
3,09/04/2021 18:23:31,0,0,0,0,0,0,0,0,False,...,False,False,False,False,False,False,False,0.0,normal,NaN
4,09/04/2021 18:23:32,0,0,0,0,0,0,0,0,True,...,False,False,False,False,False,False,False,0.0,normal,NaN


use .shape to find out the number of rows and columns in the data

In [19]:
df.shape

(7495, 44)

The data has 7495 rows and 44 columns

Use .info to get information about the 44 columns:

In [28]:
df.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7495 entries, 0 to 7494
Data columns (total 44 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Time           7495 non-null   object 
 1   Tank_1         7495 non-null   int64  
 2   Tank_2         7495 non-null   int64  
 3   Tank_3         7495 non-null   int64  
 4   Tank_4         7495 non-null   int64  
 5   Tank_5         7495 non-null   int64  
 6   Tank_6         7495 non-null   int64  
 7   Tank_7         7495 non-null   int64  
 8   Tank_8         7495 non-null   int64  
 9   Pump_1         7495 non-null   bool   
 10  Pump_2         7495 non-null   bool   
 11  Pump_3         7495 non-null   bool   
 12  Pump_4         7495 non-null   bool   
 13  Pump_5         7495 non-null   bool   
 14  Pump_6         7495 non-null   bool   
 15  Flow_sensor_1  7495 non-null   int64  
 16  Flow_sensor_2  7495 non-null   int64  
 17  Flow_sensor_3  7495 non-null   int64  
 18  Flow_sen

The columns Label_n and Lable_n are the only columns with null values. Could be a possible naming error.

Use .describe to find basic statistics for columns

In [21]:
df.describe()

,Tank_1,Tank_2,Tank_3,Tank_4,Tank_5,Tank_6,Tank_7,Tank_8,Flow_sensor_1,Flow_sensor_2,Flow_sensor_3,Flow_sensor_4,Label_n,Lable_n
count,7495.000000,7495.000000,7495.000000,7495.000000,7495.000000,7495.000000,7495.000000,7495.000000,7495.000000,7495.000000,7495.0,7495.000000,5391.000000,2104.000000
mean,729.740627,975.675384,586.644830,263.765844,367.260707,184.047899,184.227885,156.548499,1432.755170,1488.458973,0.0,1280.038426,0.317381,0.145437
std,681.092889,692.334742,673.172968,344.667758,284.188402,176.862152,208.961824,179.271719,1917.117999,1933.604049,0.0,1765.084711,0.465500,0.352625
min,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.0,0.000000,0.000000,0.000000
25%,4.000000,227.000000,75.500000,0.000000,24.500000,0.000000,0.000000,0.000000,0.000000,0.000000,0.0,0.000000,0.000000,0.000000
50%,621.000000,1106.000000,365.000000,33.000000,380.000000,149.000000,88.000000,66.000000,0.000000,0.000000,0.0,0.000000,0.000000,0.000000
75%,1334.000000,1613.000000,926.000000,470.000000,615.000000,368.000000,377.000000,324.500000,4000.000000,4000.000000,0.0,2766.500000,1.000000,0.000000
max,2021.000000,2012.000000,3427.000000,1261.000000,838.000000,800.000000,812.000000,507.000000,4000.000000,4000.000000,0.0,4789.000000,1.000000,1.000000
